Guled/put photo in add comic form

Somnibyte opened the pull request:
- CamerViewController can now successfully send back images to the AddComicViewController 
- When the AddComicViewController receives an image it applies a "apple music blur" effect. 
- Added Alert protocol that gives ViewControllers the ability to display alerts with a title, message, and a default "Ok" button. 
- Added `resize()` and `blur()` methods as an extension to UIImage. 
- Added protocol to help CamerViewController send images to the AddComicViewController.

Commit:
Add ability to apply blur. Add ability to resize images.

Commit:
Add alert protocol. Finalize apple music blur effect. Add constants to LucienConstants.

Commit:
Add new constants for AddComicViewController in LucienConstants. Implement scrollview feedback given by Paul. Redesign parts of AddComicViewController xib.

mayasaxena commented on lucien-ios/Camera/CameraViewController.swift:
```diff
@@ -9,14 +9,22 @@
 
+    // MARK: - Variables
+
+    weak var cameraViewDelegate: CameraViewDelegate?
```
If there's only one delegate in a class, you can just name it `delegate`

mayasaxena commented on lucien-ios/Camera/CameraViewController.swift:
```diff
@@ -9,14 +9,22 @@
     private var captureDevice: AVCaptureDevice?
     private let captureSessionOutput = AVCapturePhotoOutput()
-    private var didTakePhoto = false
+    private let captureOutput = AVCapturePhotoOutput()
```
Do you need both `captureSessionOutput` and `captureOutput`? They look like duplicates to me

mayasaxena commented on lucien-ios/Camera/CameraViewController.swift:
```diff
@@ -35,64 +45,68 @@ class CameraViewController: UIViewController, AVCaptureVideoDataOutputSampleBuff
+        }
+
+        guard let data = AVCapturePhotoOutput.jpegPhotoDataRepresentation(forJPEGSampleBuffer: photoSampleBuffer, previewPhotoSampleBuffer: previewPhotoSampleBuffer) else {
+            return
```
If the `else` of a `guard` is only returning, it can be a one-liner

mayasaxena commented on lucien-ios/Extensions/UIImageExtensions.swift:
```diff
@@ -0,0 +1,48 @@
+            currentFilter.setValue(radius, forKey: kCIInputRadiusKey)
+
+            if let output = currentFilter.outputImage {
+                if let cgimg = context.createCGImage(output, from: output.extent) {
```
Rename to `cgImage`

mayasaxena commented on lucien-ios/LucienConstants.swift:
```diff
@@ -9,6 +9,34 @@
+    static let textFieldBorderOffset = CGFloat(48)
+    static let pickerViewLeftimageEdgeInsetOffset = CGFloat(72)
+
+    // MARK: - Apple Music Blur Effect Constants
```
What does Apple Music have to do with these constants? If they just happen to be the same, is that information future developers will need?

mayasaxena commented on lucien-ios/Protocols/Alertable.swift:
```diff
@@ -0,0 +1,22 @@
+
+import Foundation
+
+protocol Alertable {}
```
I would consider renaming this. `Alertable` to me implies that something can be alerted - like it responds to a callback or something else similar. This is more like `AlertDisplaying`

mayasaxena commented on lucien-ios/Protocols/Alertable.swift:
```diff
@@ -0,0 +1,22 @@
+    /// Displays an alert with only one button displaying the word "Ok" along with your desired title and message."
+    func showAlert(title: String, message: String) {
+        let alertController = UIAlertController(title: title, message: message, preferredStyle: .alert)
+        let okAction = UIAlertAction(title: "Ok", style: .cancel, handler: { [weak self] _ in
```
Use trailing closure syntax here

mayasaxena commented on lucien-ios/Protocols/CameraViewDelegate.swift:
```diff
@@ -0,0 +1,13 @@
+
+import Foundation
+
+protocol CameraViewDelegate: class {
```
This doesn't need to be in a separate file since it's small and strongly associated with `CameraViewController`. It can just go in the same file as it. Also, this naming is misleading. This is not a delegate for a `CameraView`, it's one for a `CameraViewController` and should be named accordingly.

mayasaxena commented on lucien-ios/Protocols/CameraViewDelegate.swift:
```diff
@@ -0,0 +1,13 @@
+import Foundation
+
+protocol CameraViewDelegate: class {
+    func getImage(image: UIImage)
```
Naming here is a little off - take a look at `UITableViewDelegate` and others for the format. It's usually along the lines of `<view controller>Did<do action>`. In this case it should be something like `cameraViewController(viewController: CameraViewController, didCapture image: UIImage)`. Because the delegate never calls its own methods, the methods should be named like callbacks, not like actions.

Commit:
Rename CameraViewDelegate. Rename Alertable Delegate. Rename CameraViewControllerDelegate method.

IntrepidBuildServer commented on lucien-ios/AddNewComicBook/AddComicViewController.swift:
```diff
@@ -311,4 +315,35 @@ final class AddComicViewController: UIViewController, UIPickerViewDelegate, UIPi
     }
+
+    // MARK: - CameraViewDelegate
+    
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Trailing Whitespace Violation: Lines should not have trailing whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Atrailing_whitespace)

IntrepidBuildServer commented on lucien-ios/AddNewComicBook/AddComicViewController.swift:
```diff
@@ -10,10 +10,11 @@ import UIKit
 import AVFoundation
 
-final class AddComicViewController: UIViewController, UIPickerViewDelegate, UIPickerViewDataSource, UITextFieldDelegate {
+final class AddComicViewController: UIViewController, UIPickerViewDelegate, UIPickerViewDataSource, UITextFieldDelegate, CameraViewControllerDelegate, UIScrollViewDelegate {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Type Body Length Violation: Type body should span 200 lines or less excluding comments and whitespace: currently spans 218 lines [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Atype_body_length)

IntrepidBuildServer commented on lucien-ios/Camera/CameraViewController.swift:
```diff
@@ -35,64 +49,66 @@ class CameraViewController: UIViewController, AVCaptureVideoDataOutputSampleBuff
+    // MARK: - AVCapturePhotoCaptureDelegate
+
+    func photoOutput(_ captureOutput: AVCapturePhotoOutput,
+                       didFinishProcessingPhoto photoSampleBuffer: CMSampleBuffer?,
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Vertical Parameter Alignment Violation: Function parameters should be aligned vertically if they're in multiple lines in a declaration. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Avertical_parameter_alignment)

IntrepidBuildServer commented on lucien-ios/Camera/CameraViewController.swift:
```diff
@@ -35,64 +49,66 @@ class CameraViewController: UIViewController, AVCaptureVideoDataOutputSampleBuff
+
+    func photoOutput(_ captureOutput: AVCapturePhotoOutput,
+                       didFinishProcessingPhoto photoSampleBuffer: CMSampleBuffer?,
+                       previewPhoto previewPhotoSampleBuffer: CMSampleBuffer?,
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Vertical Parameter Alignment Violation: Function parameters should be aligned vertically if they're in multiple lines in a declaration. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Avertical_parameter_alignment)

IntrepidBuildServer commented on lucien-ios/Camera/CameraViewController.swift:
```diff
@@ -35,64 +49,66 @@ class CameraViewController: UIViewController, AVCaptureVideoDataOutputSampleBuff
+    func photoOutput(_ captureOutput: AVCapturePhotoOutput,
+                       didFinishProcessingPhoto photoSampleBuffer: CMSampleBuffer?,
+                       previewPhoto previewPhotoSampleBuffer: CMSampleBuffer?,
+                       resolvedSettings: AVCaptureResolvedPhotoSettings,
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Vertical Parameter Alignment Violation: Function parameters should be aligned vertically if they're in multiple lines in a declaration. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Avertical_parameter_alignment)

IntrepidBuildServer commented on lucien-ios/Camera/CameraViewController.swift:
```diff
@@ -35,64 +49,66 @@ class CameraViewController: UIViewController, AVCaptureVideoDataOutputSampleBuff
+                       didFinishProcessingPhoto photoSampleBuffer: CMSampleBuffer?,
+                       previewPhoto previewPhotoSampleBuffer: CMSampleBuffer?,
+                       resolvedSettings: AVCaptureResolvedPhotoSettings,
+                       bracketSettings: AVCaptureBracketedStillImageSettings?,
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Vertical Parameter Alignment Violation: Function parameters should be aligned vertically if they're in multiple lines in a declaration. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Avertical_parameter_alignment)

IntrepidBuildServer commented on lucien-ios/Camera/CameraViewController.swift:
```diff
@@ -35,64 +49,66 @@ class CameraViewController: UIViewController, AVCaptureVideoDataOutputSampleBuff
+                       previewPhoto previewPhotoSampleBuffer: CMSampleBuffer?,
+                       resolvedSettings: AVCaptureResolvedPhotoSettings,
+                       bracketSettings: AVCaptureBracketedStillImageSettings?,
+                       error: Error?) {
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Vertical Parameter Alignment Violation: Function parameters should be aligned vertically if they're in multiple lines in a declaration. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.intrepid.io/coding_rules#rule_key=SwiftLint%3Avertical_parameter_alignment)

IntrepidBuildServer commented:
SonarQube analysis reported 7 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 6 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

IntrepidBuildServer commented:
<!--
  0 Errors
  0 Warnings
  2 Messages: Executed 2 tests, with 0 failu...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.391 (0.393) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 8.515 (8.517) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

Commit:
Rename CameraViewDelegate MARK to CameraViewControllerDelegate.

Remove unused variables. Reformat code using extensions.

Refactored textfield return key behavior.

Prevent off by 1 error when checking textfields in textFieldShouldReturn method.

Remove tags from textfields.

Fix guard statement formatting.

ghost commented:
SonarQube analysis reported 5 issues
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

IntrepidBuildServer commented:
<!--
  1 Error: fatal error: file ‘/Users/jenk...
  0 Warnings
  2 Messages: Executed 2 tests, with 0 failu...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">fatal error: file ‘/Users/jenkins/Jenkins/workspace/os-pipeline_lucien-ios_PR-9-LZVTVRTJ3GPUW2SEMTJK22ZE3YQTH7PQWHDS3TBQAX5AGORYIJ5Q/Pods/Google/Headers/module.modulemap’ has been modified since the precompiled header ‘/Users/jenkins/Jenkins/workspace/os-pipeline_lucien-ios_PR-9-LZVTVRTJ3GPUW2SEMTJK22ZE3YQTH7PQWHDS3TBQAX5AGORYIJ5Q/DerivedData/Build/Intermediates.noindex/PrecompiledHeaders/lucien-ios-Bridging-Header-swift_11K2MF1VNJAHO-clang_2O9YIEWVALUCS.pch’ was built</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.353 (0.355) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 6.129 (6.130) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

paulrolfe commented on lucien-ios/AddNewComicBook/AddComicViewController.swift:
```diff
@@ -51,6 +51,7 @@ final class AddComicViewController: UIViewController, UIPickerViewDelegate, UIPi
 
     override func viewDidLoad() {
         super.viewDidLoad()
+        scrollView.delegate = self
```
Is this needed still?

Somnibyte replied:
It is no longer needed. I'll remove it.

paulrolfe commented on lucien-ios/AddNewComicBook/AddComicViewController.swift:
```diff
@@ -10,10 +10,11 @@ import UIKit
 import AVFoundation
 
-final class AddComicViewController: UIViewController, UIPickerViewDelegate, UIPickerViewDataSource, UITextFieldDelegate {
+final class AddComicViewController: UIViewController, UIScrollViewDelegate {
```
is the `UIScrollViewDelegate` needed still?

paulrolfe commented on lucien-ios/AddNewComicBook/AddComicViewController.swift:
```diff
@@ -276,39 +313,45 @@ final class AddComicViewController: UIViewController, UIPickerViewDelegate, UIPi
 
     func textFieldShouldReturn(_ textField: UITextField) -> Bool {
-        if let nextTextField = textField.superview?.superview?.viewWithTag(textField.tag + 1) as? UITextField {
+        if let nextTextField = textField.superview?.superview?.superview?.viewWithTag(textField.tag + 1) as? UITextField {
```
I know there wasn't a substantive change here, but this feels really fragile. I think a better solution might be to either use something like IQKeyboardManager to handle this for you. Or maybe you could keep an array of the textfields in the desired order and get the next `firstResponder` by checking that array. I think there are several good options.

Somnibyte replied:
I went with the array method. Thank you for catching that. The array option was much cleaner.

paulrolfe commented on lucien-ios/Protocols/AlertDisplaying.swift:
```diff
@@ -0,0 +1,22 @@
+import Foundation
+
+protocol AlertDisplaying {}
+extension AlertDisplaying where Self: UIViewController {
```
👍

IntrepidBuildServer commented:
<!--
  0 Errors
  0 Warnings
  2 Messages: Executed 2 tests, with 0 failu...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.345 (0.351) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 6.091 (6.092) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

IntrepidBuildServer commented:
<!--
  1 Error: fatal error: file ‘/Users/jenk...
  0 Warnings
  2 Messages: Executed 2 tests, with 0 failu...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">fatal error: file ‘/Users/jenkins/Jenkins/workspace/os-pipeline_lucien-ios_PR-9-LZVTVRTJ3GPUW2SEMTJK22ZE3YQTH7PQWHDS3TBQAX5AGORYIJ5Q/Pods/Google/Headers/module.modulemap’ has been modified since the precompiled header ‘/Users/jenkins/Jenkins/workspace/os-pipeline_lucien-ios_PR-9-LZVTVRTJ3GPUW2SEMTJK22ZE3YQTH7PQWHDS3TBQAX5AGORYIJ5Q/DerivedData/Build/Intermediates.noindex/PrecompiledHeaders/lucien-ios-Bridging-Header-swift_11K2MF1VNJAHO-clang_2O9YIEWVALUCS.pch’ was built</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.353 (0.355) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 6.272 (6.274) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

paulrolfe commented on lucien-ios/AddNewComicBook/AddComicViewController.swift:
```diff
@@ -276,39 +314,43 @@ final class AddComicViewController: UIViewController, UIPickerViewDelegate, UIPi
-            textField.resignFirstResponder()
-        }
+        guard let textFields = addComicViewControllerTextFields else { return false }
+        let nextTextField = textFields[textField.tag + 1]
```
Doesn't this crash if you hit enter on the last text field?

Either check the length before this, or use `ip_safely` subscripting from the Intrepid pod.

Also, to remove a reliance on tags from your .xib being correct, you could find the index of `textField` in the array and then grab the next one if it exists.

Somnibyte replied:
It does not crash. The reason is because the last textfield has a keypad keyboard which does not have a `return` key. It has a custom "Done" button toolbar above it that calls `resignFirstResponder`.

Somnibyte replied:
For best practice and possible crashes (edits in the future etc..) I'll change the logic to check for length.

mayasaxena replied:
Seconding Paul's comment here, we shouldn't be relying on UI (which can change) to prevent crashes, and we shouldn't be using the tags to identify text fields.

Somnibyte replied:
I tried to work with the Intrepid pod, but I decided to check for length instead due to errors like this: 

![screen shot 2017-10-10 at 4 53 53 pm](https://user-images.githubusercontent.com/3248926/31410956-20363f9a-adde-11e7-8f5a-3b90ff6447b1.png)

IntrepidBuildServer commented:
<!--
  0 Errors
  0 Warnings
  2 Messages: Executed 2 tests, with 0 failu...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.394 (0.398) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 6.764 (6.766) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

mayasaxena commented on lucien-ios/AddNewComicBook/AddComicViewController.swift:
```diff
@@ -243,15 +278,18 @@ final class AddComicViewController: UIViewController, UIPickerViewDelegate, UIPi
-               let bottomBorderSubLayer = textField.layer.sublayers,
-               let comicTitleTextFieldSubLayers = comicTitleTextField.layer.sublayers else {
-            return
+        guard let comicTitleText = comicTitleTextField.text,
```
`let` should be on its own line

mayasaxena commented on lucien-ios/AddNewComicBook/AddComicViewController.swift:
```diff
@@ -243,15 +278,18 @@ final class AddComicViewController: UIViewController, UIPickerViewDelegate, UIPi
-            return
+        guard let comicTitleText = comicTitleTextField.text,
+            let bottomBorderSubLayer = textField.layer.sublayers,
+            let comicTitleTextFieldSubLayers = comicTitleTextField.layer.sublayers else {
```
`else { return }` should be a single line indented at the same level as the `let`s

mayasaxena commented on lucien-ios/AddNewComicBook/AddComicViewController.swift:
```diff
@@ -276,39 +314,47 @@ final class AddComicViewController: UIViewController, UIPickerViewDelegate, UIPi
     func textFieldShouldReturn(_ textField: UITextField) -> Bool {
-        if let nextTextField = textField.superview?.superview?.viewWithTag(textField.tag + 1) as? UITextField {
+        guard let textFields = addComicViewControllerTextFields else { return false }
+        guard let currentTextFieldArrayIndex = textFields.index(of: textField) else { return false }
```
This assignment can be grouped with the previous `guard` since it's doing the same thing on return

mayasaxena commented on lucien-ios/AddNewComicBook/AddComicViewController.swift:
```diff
@@ -276,39 +314,47 @@ final class AddComicViewController: UIViewController, UIPickerViewDelegate, UIPi
+        guard let currentTextFieldArrayIndex = textFields.index(of: textField) else { return false }
+        let currentTextFieldIndex = textFields.startIndex.distance(to: currentTextFieldArrayIndex)
+        if currentTextFieldIndex < textFields.count {
+            let nextTextField = textFields[currentTextFieldIndex + 1]
```
I think this could still potentially crash - if you're at the last index and add one it'll be out of bounds.

mayasaxena commented on lucien-ios/AddNewComicBook/AddComicViewController.swift:
```diff
@@ -276,39 +314,47 @@ final class AddComicViewController: UIViewController, UIPickerViewDelegate, UIPi
+            let nextTextField = textFields[currentTextFieldIndex + 1]
             nextTextField.becomeFirstResponder()
-        } else {
-            textField.resignFirstResponder()
```
Why does this not need to resign anymore?

Somnibyte replied:
That was an accident. I'll bring it back.

ghost commented:
<!--
  1 Error: fatal error: file ‘/Users/jenk...
  0 Warnings
  2 Messages: Executed 2 tests, with 0 failu...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">fatal error: file ‘/Users/jenkins/Jenkins/workspace/os-pipeline_lucien-ios_PR-9-LZVTVRTJ3GPUW2SEMTJK22ZE3YQTH7PQWHDS3TBQAX5AGORYIJ5Q/Pods/Google/Headers/module.modulemap’ has been modified since the precompiled header ‘/Users/jenkins/Jenkins/workspace/os-pipeline_lucien-ios_PR-9-LZVTVRTJ3GPUW2SEMTJK22ZE3YQTH7PQWHDS3TBQAX5AGORYIJ5Q/DerivedData/Build/Intermediates.noindex/PrecompiledHeaders/lucien-ios-Bridging-Header-swift_11K2MF1VNJAHO-clang_2O9YIEWVALUCS.pch’ was built</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.419 (0.422) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 1 test, with 0 failures (0 unexpected) in 11.115 (11.118) seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>
